src/Core/Common/UseParallelizeImageRegion/Code.cxx

```diff
@@ -0,0 +1,116 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkLogImageFilter.h"
+#include "itkRandomImageSource.h"
+#include "itkImageDuplicator.h"
+#include "itkImageRegionIterator.h"
+
+constexpr unsigned int Dimension = 2;
+using PixelType = unsigned int;
+using ImageType = itk::Image< PixelType, Dimension >;
+
+//calculate log(1+x), where x is pixel value, using LogImageFilter
+void log1x_LogImageFilter(ImageType::Pointer& image)
+{
+  //LogImageFilter calculates log(x), so we have to modify the image first
+  //by increase its every pixel value by 1, and then apply log filter to it
+  itk::ImageRegionIterator<ImageType> it(image, image->GetBufferedRegion());
+  for (; !it.IsAtEnd(); ++it)
+    {
+    it.Set(1 + it.Get());
+    }
+
+  //classic filter declaration and invocation
+  using LogType = itk::LogImageFilter< ImageType, ImageType >;
+  LogType::Pointer logF = LogType::New();
+  logF->SetInput(image);
+  logF->SetInPlace(true);
+  logF->Update();
+  image = logF->GetOutput();
+  image->DisconnectPipeline();
+}
+
+//calculate log(1+x), where x is pixel value, using ParallelizeImageRegion
+void log1x_ParallelizeImageRegion(ImageType::Pointer& image)
+{
+  itk::MultiThreaderBase::Pointer mt = itk::MultiThreaderBase::New();
+  //ParallelizeImageRegion invokes the provided lambda function in parallel
+  //each invocation will contain a piece of the overall region
+  mt->ParallelizeImageRegion<Dimension>(image->GetBufferedRegion(),
```

Comment:
Again, this is more a matter of tastes, and may be a lost battle :sweat:, but the style guide states that we should leave a blank space between the template argument characters and the actual argument (`< Dimension >`). The same applies to function, control statement (`for( whatever )` instead of  `for (whatever)` etc.) arguments.

Reply:
That fights against Visual Studio's default style. Manually correcting that is super-annoying. I do try to do that when changing existing code which uses that style.

Reply:
Is there a Visual Studio ITK style file?

Reply:
In `for ( ;` there is a space before semicolon, which KWStyle does not like.

Reply:
We should consider:  http://editorconfig.org

It provides a nice way to define styles that is supported by many editors.

Hans

Reply:
ITK's style is so complicated there isn't a style tool which can enforce it in an automated and satisfactory way. Should we reconsider loosening it? Or changing it entirely, like VTK did? If so we should make it simple enough that we have some automated way of enforcing it. Should we start a Discourse discussion about this?